### ISSUE TITLE
[Bug]: "borderOpacityWhenMoving" does not take effect on the child shapes within the group.

### ISSUE THREAD

**Issue by zhe-he** [fabricjs/fabric.js#9734]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta19

### Node Version (if applicable)

None

### Link To Reproduction
https://codesandbox.io/p/devbox/fabric-borderopacitywhenmoving-ffgkr3?file=%2Fsrc%2Ftest-border-opacity.tsx%3A3%2C1

### Steps To Reproduce

1. Create a rectangle A.
2. Create a rectangle B.
3. Create a group C with members A and B.
4. Add C to the canvas.
5. Select A and drag it.

### Expected Behavior

The border of controller A should have an opacity of 0.4

### Actual Behavior

The border of controller A does not have an opacity of 0.4

[demo](https://codesandbox.io/p/devbox/fabric-borderopacitywhenmoving-ffgkr3?file=%2Fsrc%2Ftest-border-opacity.tsx%3A13%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clts9ku6100073b6gntpz19c7%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clts9ku6000023b6gazz3oiew%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clts9ku6000043b6gwbym2dly%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clts9ku6000063b6gzjlziedv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clts9ku6000023b6gazz3oiew%2522%253A%257B%2522id%2522%253A%2522clts9ku6000023b6gazz3oiew%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clts9uwli00023b6gbp9twdv1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A13%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A13%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Ftest-border-opacity.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clts9uwli00023b6gbp9twdv1%2522%257D%252C%2522clts9ku6000063b6gzjlziedv%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clts9ku6000053b6g3kej4s0l%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clts9ku6000063b6gzjlziedv%2522%252C%2522activeTabId%2522%253A%2522clts9ku6000053b6g3kej4s0l%2522%257D%252C%2522clts9ku6000043b6gwbym2dly%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clts9ku6000033b6gdaff7i6z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522clts9ku6000043b6gwbym2dly%2522%252C%2522activeTabId%2522%253A%2522clts9ku6000033b6gdaff7i6z%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by asturur** [comment]
Hi @zhe-he i agree this can be considered as a bug.
It should be easy to patch, but keep in mind this is a good case of 'a feature that doesn't make much sense anymore'.

we have 'borderOpacityWhenMoving' and a 'isMoving' boolean and a series of test to keep them working.
So when you drag an object you have an opacity on the controlbox.
But when you scale or rotate you have nothing to do so

This 'borderOpacityWhenMoving' is a relic of the past and probably something we have to put in the list of things that needs to be reconsidered.

I think for consistency sake this should be patched for now.
It won't block v6 release but should be patched regardless of what we will do in the feature with the functionality.

**Comment by zhe-he** [comment]
Because I have added auxiliary lines to my canvas, when I drag the shape, the selected border of the shape will block the auxiliary lines. So I need to set borderOpacityWhenMoving to 0, so they will not interfere. When I rotate and scale the shape, because they do not interfere strongly with the auxiliary lines, the selected border can be left unhidden.

I found in the source code that the current shape is determined to be dragged by this.isMoving. In the future, will there be this.isScaling and this.isRorating to determine if the current shape is locked and rotated? If these are added, will there also be this.isSkewing, this.isResizing...?
